src/rx/main.c

```diff
@@ -588,10 +583,13 @@ build_pattern_fsm(const struct fsm_options *opt, bool show_stats,
 		exit(EXIT_FAILURE);
 	}
 
-// TODO: therefore set endid after minimisation
-
 	assert(!fsm_empty(fsm));
 
+	if (!fsm_setendid(fsm, id)) {
```

Comment:
That there's only one end id for this fsm (as the comment just above says), so there's no need to deal with endids through the various transformations as we construct the fsm. I'm setting the id after construction just because it's unnecessary to set it any earlier.